app/assets/javascripts/discourse/app/templates/user/notifications-index.hbs

```diff
@@ -6,15 +6,22 @@
       {{i18n "errors.desc.unknown"}}
     {{/if}}
   </div>
-{{/if}}
-
-<div class="user-notifications-filter-select-kit">
-  {{notifications-filter value=filter onChange=(action (mut filter))}}
-</div>
-
-{{#if hasFilteredNotifications}}
-  {{user-notifications-large notifications=model filter=filter}}
-  {{conditional-loading-spinner condition=loading}}
+{{else if userDoesNotHaveNotifications}}
+  <div class="empty-state">
+    <span class="empty-state-title">{{i18n "user.no_notifications_page_title"}}</span>
+    <div class="empty-state-body">
+      <p>{{emptyStateBody}}</p>
+    </div>
+  </div>
 {{else}}
-  <div class="alert alert-info">{{i18n "notifications.empty"}}</div>
+  <div class="user-notifications-filter-select-kit">
+    {{notifications-filter value=filter onChange=(action (mut filter))}}
+  </div>
+
+  {{#if nothingFound}}
+    <div class="alert alert-info">{{i18n "notifications.empty"}}</div>
+  {{else}}
+    {{user-notifications-large notifications=model filter=filter}}
```

Comment:
just to be sure it's not forgotten from my other comment, what is this filter=filter doing here ?

Reply:
See the comment above.